shared/api/container_snapshot.go

```diff
@@ -20,8 +20,18 @@ type ContainerSnapshotPost struct {
 	Live bool `json:"live,omitempty" yaml:"live,omitempty"`
 }
 
+// ContainerSnapshotPut represents the modifiable fields of a LXD container snapshot
+// API extension: snapshot_expiry
+type ContainerSnapshotPut struct {
+	// Using a pointer here allows us to differentiate between missing key, empty value,
+	// and provided value. This is important when editing the snapshot config.
+	ExpiresAt *time.Time `json:"expires_at" yaml:"expires_at"`
```

Comment:
We should find a way to do this without using a pointer as that'd be inconsistent with what we do for the rest of the timestamps in the API.

Reply:
We could just allow editing the entire snapshot config. That would make things pretty simple.